assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -59,11 +61,22 @@ function useUploader(refreshLibrary = true) {
   const uploadFile = (file) => {
     // TODO Add permission check here, see Gutenberg's userCan function.
     if (!fileSizeCheck(file)) {
-      throw new Error('File size error');
+      const SizeError = new Error('File size error');
+      SizeError.name = 'SizeError';
+      SizeError.file = file.name;
+      SizeError.message = `Your file is ${bytesToMB(
+        file.size
+      )}MB and the upload limit is ${bytesToMB(
+        maxUpload
+      )}MB. Please resize and try again!`;
+      throw SizeError;
     }
 
     if (!isValidType(file)) {
-      throw new Error('File type error');
+      const ValidError = new Error('File size error');
```

Comment:
Careful, wrong error message here :)

Also, user-facing error messages need to be localized. (It wasn't before because it wasn't user-facing).

Suggestion

```js
const allowedFileTypes = Object.keys(allowedMimeTypes).reduce(
  (acc, cur) => [...acc, ...allowedMimeTypes[cur]],
  []
);

const ValidError = new Error(sprintf(
  /* translators: %s: list of allowed file extensions. */
  __('This file type is not allowed. You can only upload %s.', 'web-stories'),
  allowedFileTypes.join(', ')
));
```

Reply:
Oh, it seems like this message isn't actually used for the snackbar message? 🤔

Reply:
Yes, so basically this validation error message has some `bold` text so constructed on snackbar internally.

Reply:
Messages are localized and formatted before sent.